crates/sui-framework/packages/deepbook/sources/order_query.move

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module deepbook::order_query {
+    use std::option;
+    use std::option::{Option, some, none};
+    use std::vector;
+    use deepbook::critbit::CritbitTree;
+    use sui::linked_table;
+    use deepbook::critbit;
+    use deepbook::clob_v2;
+    use deepbook::clob_v2::{Order, Pool, TickLevel};
+
+    const PAGE_LIMIT: u64 = 100;
+
+    struct OrderPage has drop {
+        orders: vector<Order>,
+        // false if this is the last page
+        has_next_page: bool,
+        // tick level where the next page begins (if any)
+        next_tick_level: Option<u64>,
+        // order id where the next page begins (if any)
+        next_order_id: Option<u64>
+    }
+
+    // return an OrderPage starting from start_tick_level + start_order_id
+    // containing as many orders as possible without going over the
+    // dynamic fields accessed/tx limit
+    public fun iter_bids<T1, T2>(
+        pool: &Pool<T1, T2>,
+        // tick level to start from
+        start_tick_level: Option<u64>,
```

Comment:
Can we define this a little more clearly in comments, i.e. whether or not the start ticks start from the bottom of bid list (lowest priced bids, or from the top of the book highest priced bids).

In the codes case it looks like its going from bottom to top of bids.

Reply:
I am going to add a ascending bool arg to allow user to choose the order of the orders (lol)